os/net/mac/tsch/tsch-slot-operation.c

```diff
@@ -1010,12 +1011,12 @@ PT_THREAD(tsch_slot_operation(struct rtimer *t, void *ptr))
         TSCH_ASN_INC(tsch_current_asn, timeslot_diff);
         /* Time to next wake up */
         time_to_next_active_slot = timeslot_diff * tsch_timing[tsch_ts_timeslot_length] + drift_correction;
+        time_to_next_active_slot += tsch_timesync_adaptive_compensate(time_to_next_active_slot);
```

Comment:
why do we need to move this line?

Reply:
previously `tsch_schedule_slot_operation` was called with the value of `time_to_next_active_slot` before the compensation was applied.

Reply:
oh right, I somehow was `time_to_next_active_slot` as `current_slot_start` ..
got it!